src/core/zowe/core_for_zowe_sdk/credential_manager.py

```diff
@@ -0,0 +1,195 @@
+        else:
+             encoded_credential = encoded_credential.decode()
+        return encoded_credential        
+        
+    
+    @staticmethod
+    def delete_credential(service_name: str, account_name: str) -> None:
+        """
+        Delete the credential from the keyring or storage.
+        If the keyring.delete_password function is not available, iterate through and delete credentials.
+        Parameters
+        ----------
+        service_name: str
+            The service name for the credential deletion
+        account_name: str
+            The account name for the credential deletion
+        Returns
+        -------
+        None
+        """
+        
+        try:
+            keyring.delete_password(service_name, account_name)  
+        except keyring.errors.PasswordDeleteError:
+            # Handling multiple credentials stored when the operating system is Windows
+            if sys.platform == "win32":
+                # Delete the secure value with an index
+                index = 1
+                while True:
+                    field_name = f"{account_name}-{index}"
+                    try:
+                        keyring.delete_password(f"{service_name}-{index}", field_name)
+                    except keyring.errors.PasswordDeleteError:
+                        break
+                    index += 1
+    
+    @staticmethod
+    def save_secure_props()-> None:
+        """
+        Set secure_props for the given config file
+        Returns
+        -------
+        None
+        """
+        if not HAS_KEYRING:
+            return
+        
+        # Filter or suppress specific warning messages
+        warnings.filterwarnings("ignore", message="^Retrieved an UTF-8 encoded credential")
+        service_name = constants["ZoweServiceName"]
+        credential =  CredentialManager.secure_props
+        # Check if credential is a non-empty string
+        if credential:
+            is_win32 = sys.platform == "win32"
+            #  = "UTF-16" if is_win32 else "UTF-8"
+            if is_win32:
+                service_name += "/" + constants["ZoweAccountName"] 
+            
+            # Load existing credentials, if any
+            existing_credential = CredentialManager._retrieve_credential(constants["ZoweServiceName"])
+            if existing_credential:
+                    
+                # Decode the existing credential and update secure_props
+                existing_credential_bytes = base64.b64decode(existing_credential).decode()
+                existing_secure_props = commentjson.loads(existing_credential_bytes)
+                existing_secure_props.update(credential)
+                # Encode the credential
+                encoded_credential = base64.b64encode(commentjson.dumps(existing_secure_props).encode()).decode()
+                # Delete the existing credential
+                CredentialManager.delete_credential(service_name , constants["ZoweAccountName"])
```

Comment:
Previously we were storing secure properties for only a single config file in memory, and had to rebuild the entire object. Now that CredentialManager stores the entire secure properties object, we don't need to load the existing credential anymore.

Therefore I think these lines can be removed 🙂 Except on Windows, we should delete the credential if it already exists, to clean up old properties that may have a different name (like `Zowe/secure_config_props-1`). This logic could be added to the `if is_win32` block above.